src/mpid/ch3/include/mpidpre.h

```diff
@@ -722,6 +722,7 @@ int MPID_Comm_get_lpid(MPIR_Comm *comm_ptr, int idx, int * lpid_ptr, bool is_rem
 void MPID_Request_create_hook(MPIR_Request *);
 void MPID_Request_free_hook(MPIR_Request *);
 void MPID_Request_destroy_hook(MPIR_Request *);
+void MPID_Request_set_completed(MPIR_Request *);
```

Comment:
I'm not sure why this function has ADI namespace. It looks like a ch4 internal function. I think this is a mistake to add it to ch3.

Reply:
Before it seems only being used by ch4 internally. But now I do need an ADI function for MPIR-layer requests.

`MPID_Request_complete` is called inside per-vci critical section, so MPIR-layer cannot use that, or it will end up with either missed lock or recursive locking.

I guess the alternative is to swap `MPID_Request_complete` and `MPID_Request_set_completed`, make the former for `MPIR-`layer to use and the latter internal. It will be a much bigger code churn for ch4 though. (ch3 don't have to change since it does need differentiate).